proposals/0029-cooperative-vector.md

```diff
@@ -102,383 +113,325 @@ Introduce new HLSL intrinsics to accelarate matrix-vector operations. In this sp
+    immarg i32)       ; matrix layout 
 ```
-// Declare an input vector
-vector<uint, 8> ipVector;
-
-// Set interpretation value to DXILCoopVectorTypeInterpretation::SignedInt8x4Packed
-// Each uint element (32-bit) in the input vector, ipVector, will be interpreted as 4 int8 values in the VectorMatrixMul intrinsic. 
-// Note that InputTy = uint and InputComponents = 8 (8 x 4 = 32 sint8 values )
-VecMatOpDescriptor<32 /*M*/, 
-                   32 /*K*/, 
-                   DXILTypeInterpretation::SignedInt8x4Packed /*InputInterpretation*/, 
-                   DXILTypeInterpretation::SignedInt8 /*MatrixInterpretation*/,
-                   DXILTypeInterpretation::Unsupported /*BiasInterpretation*/, 
-                   DXILMatrixLayout::InferencingOptimal /*Layout*/,
-                   false /*Transpose*/> desc;
-
-vector<int, 32> resultVector; //Note that the ResultComponents equals M(32)
-// Matrix is a ByteAddressBuffer
-resultVector = VectorMatrixMul(ipVector, Matrix, 0/*MatrixOffset*/, 0/*MatrixStride*/);
 
-```
+#### Overview
 
-Non-Packed Case:
+Computes the outer product between column vectors and an **M**x**N** matrix is
+accumulated component-wise atomically (with device scope) in memory. 
+
+``` 
+ResultMatrix = InputVector1 * Transpose(InputVector2); 
 ```
-// Declare an input vector
-vector<float, 32> ipVector;
 
-// Set interpretation value to DXILCoopVectorTypeInterpretation::SignedInt8x4Packed
-// Each float element of the input vector, ipVector, will be arithmetically converted to a sint8 value in the VectorMatrixMul intrinsic. 
-VecMatOpDescriptor<64 /*M*/, 
-                   32 /*K*/, 
-                   DXILTypeInterpretation::SignedInt8 /*InputInterpretation*/, 
-                   DXILTypeInterpretation::SignedInt8 /*MatrixInterpretation*/,
-                   DXILTypeInterpretation::SignedInt8 /*BiasInterpretation*/, 
-                   DXILMatrixLayout::InferencingOptimal /*Layout*/,
-                   false /*Transpose*/> desc;
 
-vector<int, 64> resultVector; // Note that the ResultComponents equals M(64)
+#### Arguments
 
-// Matrix and Bias are ByteAddressBuffers
-resultVector = VectorMatrixMul(ipVector, Matrix, 0/*MatrixOffset*/, 0/*MatrixStride*/, Bias, 0/*BiasStride*/);
+The two input vectors are specified via **input vector 1** and **input vector
+2**.
 
-```
+The matrix is accumulated to the writeable raw-buffer specified by **matrix
+resource**, with **matrix offset**, **matrix stride**, **matrix interpretation**
+and **matrix layout** behaving as described
+[above](#matrix-vector-multiply-and-multiply-add-operations).
 
+The base address of **matrix resource** and **matrix offset** must be 64 byte
+aligned.
 
-**Vector Outer Product**
+Not all combinations of vector element type and matrix interpretations are
+supported by all implementations. [CheckFeatureSupport] can be used to determine
+which combinations. A list of combinations that are guaranteed to be supported
```

Comment:
which combinations are supported and/ or not supported? The sentence is incomplete.

Reply:
changed to "...determine which combinations are supported."

(The API returns a list of supported combinations.)